src/reducers-actions/trades.ts

```diff
@@ -29,30 +39,99 @@ export const updateLastCheckedBlock = (lastCheckedBlock: number): UpdateBlockAct
 })
 
 // TODO: store to/load from localStorage
-export const INITIAL_TRADES_STATE = { trades: [] }
+export const INITIAL_TRADES_STATE = { trades: [], reverts: [] }
+
+function groupByRevertKey<T extends EventWithBlockInfo>(list: T[], map: Map<string, T[]>): void {
+  list.forEach(item => {
+    if (map.has(item.revertKey)) {
+      // Do I have to pop it out?
+      const subList = map.get(item.revertKey) as T[]
+
+      // Do not insert duplicates
+      if (!subList.find(({ id }) => item.id === id)) {
+        subList.push(item)
+        // Should I sort on insert?
+        subList.sort((a, b) => a.timestamp - b.timestamp)
+        // Do I have to put it back in?
+        map.set(item.revertKey, subList)
+      }
+    } else {
+      map.set(item.revertKey, [item])
+    }
+  })
+}
+
+function flattenGroup<T extends EventWithBlockInfo>(map: Map<string, T[]>): T[] {
+  return Array.from(map.keys()).reduce<T[]>((acc, key) => {
+    const list = map.get(key) as T[]
+    return acc.concat(list)
+  }, [])
+}
+
+function applyRevertsToTrades(trades: Trade[], reverts: TradeReversion[]): [Trade[], TradeReversion[]] {
+  const tradesByRevertKey = new Map<string, Trade[]>()
+  const revertsByRevertKey = new Map<string, TradeReversion[]>()
+
+  // Group trades by revertKey
+  groupByRevertKey(trades, tradesByRevertKey)
```

Comment:
it's not super good practice to mutate things inside a method. Imo, groupByRevertKey would need to receive only the trades, and do a "reduce" operation starting the reduce with `new Map<string, Trade[]>()`